src/main/java/edu/cornell/kfs/coa/document/validation/impl/SubAccountGlobalRule.java

```diff
@@ -444,13 +457,371 @@ protected boolean checkCgIcrRules(String subAccountTypeCode, SubAccountGlobalDet
+        return StringUtils.isBlank(subAccountGlobal.getNewSubAccountTypeCode())
+                && StringUtils.isBlank(subAccountGlobal.getNewSubAccountName())
+                && StringUtils.isBlank(subAccountGlobal.getNewSubAccountNumber())
+                && !subAccountGlobal.isNewSubAccountOffCampusCode()
+                && !subAccountGlobal.isApplyToAllNewSubAccounts()
+                && subAccountGlobal.getSubAccountGlobalNewAccountDetails().isEmpty();
+    }
+    
+    protected boolean checkBasicGlobalFieldsAndEditSubAccountSectionAreEmpty(SubAccountGlobal subAccountGlobal) {
+        boolean success = true;
+        if (StringUtils.isNotBlank(subAccountGlobal.getSubAccountName())) {
+            putFieldErrorWithDisplayNameAddedToMessage(
+                    KFSPropertyConstants.SUB_ACCOUNT_NAME, CUKFSKeyConstants.ERROR_DOCUMENT_SUB_ACCOUNT_GLOBAL_INVALID_PROPERTY_FOR_NEW_SUB_ACCOUNT);
+            success = false;
+        }
+        if (subAccountGlobal.isInactivate()) {
+            putFieldErrorWithDisplayNameAddedToMessage(
+                    CUKFSPropertyConstants.INACTIVATE, CUKFSKeyConstants.ERROR_DOCUMENT_SUB_ACCOUNT_GLOBAL_INVALID_PROPERTY_FOR_NEW_SUB_ACCOUNT);
+            success = false;
+        }
+        if (subAccountGlobal.getA21SubAccount().isOffCampusCode()) {
+            putFieldErrorWithDisplayNameAddedToMessage(
+                    KFSPropertyConstants.A21_SUB_ACCOUNT + KFSConstants.DELIMITER + KFSPropertyConstants.OFF_CAMPUS_CODE,
+                    CUKFSKeyConstants.ERROR_DOCUMENT_SUB_ACCOUNT_GLOBAL_INVALID_PROPERTY_FOR_NEW_SUB_ACCOUNT);
+            success = false;
+        }
+        if (!subAccountGlobal.getSubAccountGlobalDetails().isEmpty()) {
+            putFieldError(CUKFSPropertyConstants.SUB_ACCOUNT_GLBL_CHANGE_DETAILS,
+                    CUKFSKeyConstants.ERROR_DOCUMENT_SUB_ACCOUNT_GLOBAL_CANNOT_SPECIFY_CHANGES_AND_ADDITIONS);
+            success = false;
+        }
+        return success;
+    }
+    
+    protected boolean checkSubAccountTypeIsSpecified(SubAccountGlobal subAccountGlobal) {
+        if (StringUtils.isBlank(subAccountGlobal.getNewSubAccountTypeCode())) {
+            putFieldErrorWithDisplayNameAddedToMessage(CUKFSPropertyConstants.NEW_SUB_ACCOUNT_TYPE_CODE,
+                    CUKFSKeyConstants.ERROR_DOCUMENT_SUB_ACCOUNT_GLOBAL_REQUIRED_FIELD_FOR_NEW_SUB_ACCOUNT);
+            return false;
+        }
+        return true;
+    }
+    
+    protected boolean checkAccountsExist(SubAccountGlobal subAccountGlobal) {
+        boolean success = true;
+        int i = 0;
+        for (SubAccountGlobalNewAccountDetail newAccountDetail : subAccountGlobal.getSubAccountGlobalNewAccountDetails()) {
+            newAccountDetail.refreshReferenceObject(KFSPropertyConstants.ACCOUNT);
+            if (ObjectUtils.isNull(newAccountDetail.getAccount())) {
+                String propertyName = buildListObjectPropertyPath(
```

Comment:
should this also check if the account is active?

Reply:
I'm not finding any existing active-account validation for this doc except for the ICR accounts, so I'm not sure if an active-account check is needed for the new-sub-account lines.

Reply:
OK, thanks for looking